django_components/templatetags/component_tags.py

```diff
@@ -38,16 +39,43 @@ class TokenType:
 COMPONENT_CONTEXT_KEY = "component_context"
 
 
-@register.simple_tag(name="component_dependencies")
-def component_dependencies_tag():
+def iter_components_from_registry(registry):
+    """Yields unique components from the registry."""
+
     unique_component_classes = set(registry.all().values())
 
-    out = []
     for component_class in unique_component_classes:
-        component = component_class()
-        out.append(component.render_dependencies())
+        yield component_class()
```

Comment:
I would prefer to keep using good old for loops here instead of iterators and generators. This makes the project more approachable to beginners, and I don't think we gain anything by using them in this case.

Reply:
Sure no problem.